Util/PasswordUpdater.php

```diff
@@ -34,13 +34,15 @@ public function hashPassword(UserInterface $user)
     {
         $plainPassword = $user->getPlainPassword();
 
-        if (0 === strlen($plainPassword)) {
+        if ('' === $plainPassword) {
             return;
         }
 
         $encoder = $this->encoderFactory->getEncoder($user);
 
-        if ($encoder instanceof BCryptPasswordEncoder || $encoder instanceof SelfSaltingEncoderInterface) {
+        if ($encoder instanceof SelfSaltingEncoderInterface ||
+            (class_exists(BCryptPasswordEncoder::class) && $encoder instanceof BCryptPasswordEncoder)
```

Comment:
no need for `class_exists`. `instanceof nonExistentClass` is fine (and won't even need to trigger autoloading).

Reply:
Ok. Change it